frontend/public/components/nav/items.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { Link, LinkProps } from 'react-router-dom';
+import * as _ from 'lodash-es';
+import { NavItem } from '@patternfly/react-core';
+
+import { formatNamespacedRouteForResource } from '../../actions/ui';
+import { referenceForModel, K8sKind } from '../../module/k8s';
+import { stripBasePath } from '../utils';
+
+export const matchesPath = (resourcePath, prefix) => resourcePath === prefix || _.startsWith(resourcePath, `${prefix}/`);
+export const matchesModel = (resourcePath, model) => model && matchesPath(resourcePath, referenceForModel(model));
+
+export const stripNS = href => {
+  href = stripBasePath(href);
+  return href.replace(/^\/?k8s\//, '').replace(/^\/?(cluster|all-namespaces|ns\/[^/]*)/, '').replace(/^\//, '');
+};
+
+export const ExternalLink = ({href, name}) => <NavItem isActive={false}>
+  <a className="pf-c-nav__link" href={href} target="_blank" rel="noopener noreferrer">{name}<span className="co-external-link"></span></a>
+</NavItem>;
+
+class NavLink<P extends NavLinkProps> extends React.PureComponent<P> {
+  static defaultProps = {
+    required: '',
+    disallowed: '',
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+  static isActive(...args): boolean {
+    throw new Error('not implemented');
+  }
+
+  get to(): string {
+    throw new Error('not implemented');
+  }
+
+  static startsWith(resourcePath: string, ...someStrings: string[]) {
+    return _.some(someStrings, s => resourcePath.startsWith(s));
+  }
+
+  render() {
+    const { isActive, id, name, onClick } = this.props;
+
+    // onClick is now handled globally by the Nav's onSelect,
+    // however onClick can still be passed if desired in certain cases
+
+    return (
+      <NavItem isActive={isActive}>
+        <Link
+          id={id}
+          to={this.to}
+          onClick={onClick}
+        >
+          {name}
+        </Link>
+      </NavItem>
+    );
+  }
+}
+
+export class ResourceNSLink extends NavLink<ResourceNSLinkProps> {
```

Comment:
Not new, but I'd like to get away from using inheritance in the nav components. Not necessary for this PR, just an observation.

https://reactjs.org/docs/composition-vs-inheritance.html